presto-cli/src/main/java/com/facebook/presto/cli/Query.java

```diff
@@ -167,7 +167,13 @@ private void renderUpdate(PrintStream out, QueryResults results)
     private void discardResults()
     {
         try (OutputHandler handler = new OutputHandler(new NullPrinter())) {
-            handler.processRows(client);
+            try {
+                handler.processRows(client);
+            }
+            catch (IOException e) {
+                handler.fail();
```

Comment:
Why not call this in the outer catch block? I don't think the inner try/catch is needed